### PR TITLE
Add DOM.jest.js test

### PR DESCRIPTION
**Pull request by damassi**
@erikdstock - try this -- I converted it to jest and am avoiding react completely; Sometimes its best to just think about it in terms of JS classes and stub out all of the deps you don't need. 

Where that console log is you'll see things working; can spy / stub (`jest.fn()`) things out to ensure certain flows are being called

### CODE REVIEW

File: src/desktop/apps/auction/components/__tests__/DOM.jest.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { test } from '../DOM'
+const { DOM } = test
```
Comment (damassi) [inline src/desktop/apps/auction/components/__tests__/DOM.jest.js:2]: Get at the component, not the wrapped component

File: src/desktop/apps/auction/components/__tests__/DOM.jest.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+jest.mock('jquery', x => () => ({ foo: 'bar' }))
+jest.mock('desktop/lib/mediator.coffee', x => x => x)
+
+describe('DOM Interactions', () => {
+  describe('.handleRegister', () => {
+    it('opens the login modal if there is no user', () => {
+      DOM.prototype.componentDidMount = jest.fn()
```
Comment (damassi) [inline src/desktop/apps/auction/components/__tests__/DOM.jest.js:10]: zap out things you don't want called / don't exactly need to test

File: src/desktop/apps/auction/components/DOM.js [hunk 3]
```diff
@@ -65,6 +66,7 @@ class DOM extends Component {
       // If the user already has a CC, show accept conditions
       // (which redirects to auction-registration/:slug)
     } else if (me.has_qualified_credit_cards) {
+      console.log('this is working...')
```
Comment (damassi) [inline src/desktop/apps/auction/components/DOM.js:69]: this branch is executing when running `yarn jest --watch`

File: src/desktop/apps/auction/__tests__/utils/renderTestComponent.js [hunk 4]
```diff
@@ -1,4 +1,8 @@
-import 'jsdom-global/register'
+// Include JSDom if running from Mocha. Jest automatically injects.
```
Comment (erikdstock) [inline src/desktop/apps/auction/__tests__/utils/renderTestComponent.js:1]: 🤙🏻

**Review by erikdstock — COMMENTED** [review]
Thanks!

**Pull request merged**